lib/kafka_producer.js

```diff
@@ -245,6 +285,44 @@ KafkaProducer.prototype.flushEntireCache = function flushEntireCache(callback) {
     }
 };
 
+KafkaProducer.prototype._auditNewMsg = function _auditNewMsg(topic, msgCount, callback) {
+    var self = this;
+
+    if (!self.topicToMsgcntMap[topic]) {
+        self.topicToMsgcntMap[topic] = msgCount;
+    } else {
+        self.topicToMsgcntMap[topic] += msgCount;
+    }
+};
+
+KafkaProducer.prototype._generateAuditMsg = function _generateAuditMsg(callback) {
+    var self = this;
+    var keys = Object.keys(self.topicToMsgcntMap);
+    if (keys.length > 0) {
+        var timeBeginInSec = Math.floor((Date.now() / 1000) / self.auditTimeBucketIntervalInSec) *
+                             self.auditTimeBucketIntervalInSec;
+        var auditMsg = '{' +
```

Comment:
Replace this with something like this:

var auditMsg = {
	 time_bucket_start: timeBeginInSec,
 	 time_bucket_end: timeBeginInSec + self.auditTimeBucketIntervalInSec,            tier: self.auditTier,
   hostname:hostName,
   datacenter: self.auditDatacenter,
   topic_count: {}
};

for(topic in keys) {
	auditMsg.topic_count[topic] = self.topicToMsgcntMap[topic];
}

return JSON.stringify(auditMsg);